### PR TITLE
[imgui] Fix deps of feature allegro5-binding 

### LINKED ISSUE #17058: [imgui] fatal error: 'imgui_impl_allegro5.h' file not found
**Is your feature request related to a problem? Please describe.**

### `CMakeLists.txt`
```cmake
cmake_minimum_required(VERSION 3.0)
cmake_policy(SET CMP0048 NEW)
project(pp VERSION 0.0.0 LANGUAGES CXX)
set(CMAKE_CXX_STANDARD 14)

set(src src/main.cpp)

find_package(imgui CONFIG REQUIRED)
add_executable(${PROJECT_NAME} ${src})
target_link_libraries(${PROJECT_NAME} PRIVATE imgui::imgui)
````

### `src/main.cpp`
https://raw.githubusercontent.com/ocornut/imgui/6a161b8/examples/example_allegro5/main.cpp

Renamed the header to `#include <imgui.h>` and `#include <imgui_impl_allegro5.h>`. Also tried `#include "imgui_impl_allegro5.h"` for the latter…

### `shell`
```sh
$ [path]/vcpkg install allegro5 imgui
$ mkdir build && cd $_
$ cmake -DCMAKE_TOOLCHAIN_FILE=[root]/vcpkg/scripts/buildsystems/vcpkg.cmake ..
$ cmake --build .
[ 50%] Building CXX object CMakeFiles/pp.dir/src/main.cpp.o
/tmp/a/src/main.cpp:9:10: fatal error: 'imgui_impl_allegro5.h' file not found
#include "imgui_impl_allegro5.h"
         ^~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
make[2]: *** [CMakeFiles/pp.dir/src/main.cpp.o] Error 1
make[1]: *** [CMakeFiles/pp.dir/all] Error 2
make: *** [all] Error 2
```

**Proposed solution**
Would be nice if the example would build

**Describe alternatives you've considered**
I'm trying other cross-platform GUI frameworks that vcpkg 'supports', e.g., see the trackbacks at https://github.com/microsoft/vcpkg/issues/17052

BTW: I also tried manually configuring it, like so:
```cmake
cmake_minimum_required(VERSION 3.0)
cmake_policy(SET CMP0048 NEW)
project(pp VERSION 0.0.0 LANGUAGES CXX)
set(CMAKE_CXX_STANDARD 14)

set(src src/main.cpp)

# The `imgui_impl_allegro5.h` header was missing
file(DOWNLOAD https://raw.githubusercontent.com/ocornut/imgui/6a161b8/backends/imgui_impl_allegro5.h download_headers/imgui_impl_allegro5.h
     EXPECTED_HASH SHA256=d7d015f196ed87c5099d62f3782e321a113b057f830bfa5e292117632e4d59a5)
include_directories(${CMAKE_CURRENT_BINARY_DIR}/downloaded_headers)

find_package(imgui CONFIG REQUIRED)
add_executable(${PROJECT_NAME} ${src})
target_link_libraries(${PROJECT_NAME} PRIVATE imgui::imgui)
```

But it could find neither `#include "imgui_impl_allegro5.h"` nor `#include <imgui_impl_allegro5.h>`…

### PR DESCRIPTION
**Pull request by LilyWangLL**
Fixes #17058

1. Fix include property for `allegro5` targets.
2. Fix deps `allegro5` of `imgui[allegro5-binding]`.

Usage test pass on `x64-linux`.
The test codes from https://github.com/microsoft/vcpkg/pull/17125#issuecomment-815485360 passed on `x64-linux`.

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [ ] ~~SHA512s are updated for each updated download~~
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Review by Cheney-W — APPROVED** [review]

File: ports/imgui/CMakeLists.txt [hunk 1]
```diff
@@ -31,8 +31,8 @@ target_sources(
 target_compile_features(${PROJECT_NAME} PUBLIC cxx_std_11)
 
 if(IMGUI_BUILD_ALLEGRO5_BINDING)
-    find_path(ALLEGRO5_INCLUDE_DIRS allegro5/allegro.h)
-    target_include_directories(${PROJECT_NAME} PRIVATE ${ALLEGRO5_INCLUDE_DIRS})
+    find_package(unofficial-allegro5 CONFIG REQUIRED)
+    target_link_libraries(${PROJECT_NAME} PRIVATE unofficial-allegro5::allegro unofficial-allegro5::allegro_ttf unofficial-allegro5::allegro_font unofficial-allegro5::allegro_main)
```
Comment (dg0yt) [inline ports/imgui/CMakeLists.txt:35]: These target don't use the proper `unofficial::allegro5::` namespace.
IIUC no targets were exported by allegro5 until now, so the time to fix allegro5 is now.

Comment (dg0yt) [reply ports/imgui/CMakeLists.txt:35]: BTW are all these link libraries needed here? `unofficial-allegro5::allegro_main` doesn't add another `main` function?

Comment (SamuelMarks) [reply ports/imgui/CMakeLists.txt:35]: I think this got merged to master already

Comment (dg0yt) [reply ports/imgui/CMakeLists.txt:35]: I know, but it is only a few hours, and such mistakes have been fixed in the path. 
It really sucks if there is neither enough time for review, nor enough quality.

**Pull request merged**